RELEASE.md

```diff
@@ -0,0 +1,23 @@
+# Releasing a new `wit-bindgen-go` version
```

Comment:
Maybe just call this `# Release` since it applies to the whole Go module, and all packages within.